### PR TITLE
Pb fields preparer

### PR DESCRIPTION
**Pull request by darkushin**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-5824

## Description
Convert playbook access fields from incident to alert.

### CODE REVIEW

File: demisto_sdk/commands/content_graph/objects/modeling_rule.py [hunk 1]
```diff
@@ -14,12 +14,12 @@ def metadata_fields(self) -> Set[str]:
 
     def prepare_for_upload(
         self,
-        marketplace: MarketplaceVersions = MarketplaceVersions.MarketplaceV2,
+        current_marketplace: MarketplaceVersions = MarketplaceVersions.XSOAR,
```
Comment (dorschw) [inline demisto_sdk/commands/content_graph/objects/modeling_rule.py:17]: default xsoar?

File: demisto_sdk/commands/prepare_content/preparers/incident_to_alert.py [hunk 2]
```diff
@@ -89,3 +90,31 @@ def edit_names_and_descriptions_for_playbook(
...
+
+def replace_playbook_access_fields_recursively(datum: Any) -> Any:
+
+    if isinstance(datum, list):
+        return [replace_playbook_access_fields_recursively(item) for item in datum]
+
+    elif isinstance(datum, dict):
+        for key, val in datum.items():
+            if isinstance(val, str):
+                if key in ["root", "simple"] and "incident" in val:
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/preparers/incident_to_alert.py:103]:
```suggestion
                if key in {"root", "simple"} and "incident" in val:
```

Comment (dorschw) [inline demisto_sdk/commands/prepare_content/preparers/incident_to_alert.py:103]: does it handle `Incident`/`INCIDENT`? Do we want to handle those cases?

File: demisto_sdk/commands/prepare_content/preparers/marketplace_incident_to_alert_playbooks_prepare.py [hunk 3]
```diff
@@ -12,7 +13,8 @@ class MarketplaceIncidentToAlertPlaybooksPreparer:
     @staticmethod
     def prepare(
         data: dict,
-        marketplace: MarketplaceVersions = MarketplaceVersions.XSOAR,
+        current_marketplace: MarketplaceVersions = MarketplaceVersions.XSOAR,
+        supported_marketplaces=None,
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/preparers/marketplace_incident_to_alert_playbooks_prepare.py:17]: please add a type hint

File: demisto_sdk/commands/prepare_content/preparers/marketplace_incident_to_alert_playbooks_prepare.py [hunk 4]
```diff
@@ -21,12 +23,25 @@ def prepare(
         In any case (for all Marketplaces) remove the wrapper (<-incident-> to incident, <-incidents-> to incidents).
         Args:
             data: content item data
-            marketplace: Marketplace.
+            supported_marketplaces: list of the marketplaces this content item supports.
+            current_marketplace: Marketplace.
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/preparers/marketplace_incident_to_alert_playbooks_prepare.py:27]:
```suggestion
            current_marketplace: the destination marketplace.
```

File: demisto_sdk/commands/prepare_content/preparers/marketplace_incident_to_alert_playbooks_prepare.py [hunk 5]
```diff
@@ -21,12 +23,25 @@ def prepare(
...
         Returns: A (possibly) modified content item data
 
         """
 
-        data = prepare_descriptions_and_names(data, marketplace)
+        if supported_marketplaces is None:
+            supported_marketplaces = [
+                MarketplaceVersions.XSOAR,
+                MarketplaceVersions.MarketplaceV2,
+            ]
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/preparers/marketplace_incident_to_alert_playbooks_prepare.py:37]:
```suggestion
            supported_marketplaces = list(MarketplaceVersions)
```

File: demisto_sdk/commands/prepare_content/preparers/marketplace_incident_to_alert_playbooks_prepare.py [hunk 6]
```diff
@@ -21,12 +23,25 @@ def prepare(
...
-            marketplace: Marketplace.
+            supported_marketplaces: list of the marketplaces this content item supports.
+            current_marketplace: Marketplace.
 
         Returns: A (possibly) modified content item data
 
         """
 
-        data = prepare_descriptions_and_names(data, marketplace)
+        if supported_marketplaces is None:
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/preparers/marketplace_incident_to_alert_playbooks_prepare.py:33]: please document, in an inline comment, when this case occurs

File: demisto_sdk/commands/prepare_content/preparers/marketplace_incident_to_alert_playbooks_prepare.py [hunk 7]
```diff
@@ -21,12 +23,25 @@ def prepare(
...
+            supported_marketplaces = [
+                MarketplaceVersions.XSOAR,
+                MarketplaceVersions.MarketplaceV2,
+            ]
+        data = prepare_descriptions_and_names(data, current_marketplace)
+
+        if (
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/preparers/marketplace_incident_to_alert_playbooks_prepare.py:40]: please add a comment explaining this logic

File: demisto_sdk/commands/prepare_content/preparers/marketplace_suffix_preparer.py [hunk 8]
```diff
@@ -14,21 +14,28 @@ class MarketplaceSuffixPreparer:
...
+            current_marketplace: Marketplace. Used to determine the specific suffix
 
         Returns: A (possibliy) modified content item data
 
         """
+        if supported_marketplaces is None:
+            supported_marketplaces = [
+                MarketplaceVersions.XSOAR,
+                MarketplaceVersions.MarketplaceV2,
+            ]
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/preparers/marketplace_suffix_preparer.py:35]:
```suggestion
            supported_marketplaces = list(MarketplaceVersions)
```

File: demisto_sdk/commands/prepare_content/tests/marketplace_incident_to_alert_playbooks_prepare_test.py [hunk 9]
```diff
@@ -14,12 +14,17 @@ def test_marketplace_version_is_xsiam():
     Given:
         - A playbook which contains in its second task description the word incident, and in its third task name and
          description the word <-incident->. Also in the external description, contains three times the word incident.
-        - MarketplaceVersions.MarketplaceV2 (Which is XSIAM Marketplace version)
+         Additionally, multiple access fields that should be replaced from incident to alert.
+        - MarketplaceVersions.MarketplaceV2 (Which is XSIAM Marketplace version) as the current marketplace.
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/tests/marketplace_incident_to_alert_playbooks_prepare_test.py:18]:
```suggestion
        - MarketplaceVersions.MarketplaceV2 as the current marketplace.
```

File: demisto_sdk/commands/prepare_content/tests/marketplace_incident_to_alert_playbooks_prepare_test.py [hunk 10]
```diff
@@ -50,18 +60,36 @@ def test_marketplace_version_is_xsiam():
...
+    )
+    assert data["outputs"][0]["contextPath"] == "incident.fieldname"
+
 
 def test_marketplace_version_is_xsiam_2():
     """
     Given:
-        - A playbook which contains in the external description, contains five times the word incident/s with or without
+        - A playbook which contains in the external description five times the word incident/s with or without
          the wrapper.
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/tests/marketplace_incident_to_alert_playbooks_prepare_test.py:84]:
```suggestion
        - A playbook which contains the word incident/s (with or without a wrapper) in the external description five times.
```

**Review by dorschw — COMMENTED** [review]
Very cool

File: demisto_sdk/commands/content_graph/objects/modeling_rule.py (hunk 1 above)
Comment (darkushin) [reply demisto_sdk/commands/content_graph/objects/modeling_rule.py]: Nice catch! Changed it back...

File: demisto_sdk/commands/prepare_content/preparers/incident_to_alert.py (hunk 2 above)
Comment (darkushin) [reply demisto_sdk/commands/prepare_content/preparers/incident_to_alert.py]: No it doesn't. As far as I understand, we want to change only the `incident` access field, all other variations that might appear (e.g. input/output keys, etc.) shouldn't be changed. I plan to verify this again with Anar in the demo.

File: demisto_sdk/commands/prepare_content/preparers/marketplace_incident_to_alert_playbooks_prepare.py (hunk 3 above)
Comment (darkushin) [reply demisto_sdk/commands/prepare_content/preparers/marketplace_incident_to_alert_playbooks_prepare.py:17]: This should be a list with the default value of `list(MarketplaceVersions)`, but setting this here will raise a `Default argument value is mutable `. Therefore, I set it here with None and place the default values in the first line in the function. Do you know a better way?

**Comment by dorschw** [comment]
merged in #3025

**Pull request closed without merging**